### PR TITLE
ecl: update to 24.5.10

### PR DESCRIPTION
**Pull request by hoanga**
this updates ecl to release 24.5.10

```
> uname -a
Haiku shredder 1 hrev58146 Sep 18 2024 06:37:00 x86_64 x86_64 Haiku

> ecl
ECL (Embeddable Common-Lisp) 24.5.10 (git:3b9071f4ecc7052d5a58ba9dbe590a41615bbdc5)
Copyright (C) 1984 Taiichi Yuasa and Masami Hagiya
Copyright (C) 1993 Giuseppe Attardi
Copyright (C) 2013 Juan J. Garcia-Ripoll
Copyright (C) 2018 Daniel Kochmanski
Copyright (C) 2023 Daniel Kochmanski and Marius Gerbershagen
ECL is free software, and you are welcome to redistribute it
under certain conditions; see file 'Copyright' for details.
Type :h for Help.
Top level.
> (+ 2 2)
4
>
```

### CODE REVIEW

**Comment by Begasus** [comment]
This should be split into a base and devel package looking at it's content.

File: dev-lisp/ecl/ecl-24.5.10.recipe [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+HOMEPAGE="https://www.ecl.org"
+COPYRIGHT="2015 Daniel Kochmański
+	2000 Juan Jose Garcia Ripoll
+	1990, 1991, 1993 Giuseppe Attardi
+	1984 Taiichi Yuasa and Masami Hagiya"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="https://common-lisp.net/project/ecl/static/files/release/ecl-$portVersion.tgz"
+CHECKSUM_SHA256="e4ea65bb1861e0e495386bfa8bc673bd014e96d3cf9d91e9038f91435cbe622b"
+PATCHES="ecl-24.5.10.patchset"
```
Comment (Begasus) [inline dev-lisp/ecl/ecl-24.5.10.recipe:15]:
```suggestion
PATCHES="ecl-$portVersion.patchset"
```

**Comment by Begasus** [comment]
Also remove the previous version?

File: dev-lisp/ecl/ecl-24.5.10.recipe (hunk 1 above)
Comment (hoanga) [reply dev-lisp/ecl/ecl-24.5.10.recipe]: updated and removed previous version along with attempt to split base and devel packages

Comment (Begasus) [reply dev-lisp/ecl/ecl-24.5.10.recipe]: Will run some checks, thanks!

**Comment by Begasus** [comment]
At the right path, some changes needed:
Remove devl:libecl from PROVIDES (it's now provided in PROVIDES_devel)
Move cmd:ecl-config and it's manpage to the devel package also (there is no pkgconfig file, so the fixPkgconfig can be removed also)

Is the cmd:ecl actually used or is the library the main use? If so maybe move the cmd:ecl to a seperate tools packge?

File: dev-lisp/ecl/ecl-24.5.10.recipe [hunk 2]
```diff
@@ -66,6 +77,13 @@ INSTALL()
 {
 	make install
 
+	prepareInstalledDevelLib \
+		libecl
+	fixPkgconfig
+
+	packageEntries devel \
+		$developDir
+
```
Comment (Begasus) [inline dev-lisp/ecl/ecl-24.5.10.recipe:86]: You can enable 32bit (SECONDARY_ARCHITECTURES) for this, just change "$prefix/lib" into "$libDir" in the next part.

**Comment by hoanga** [comment]
> At the right path, some changes needed: Remove devl:libecl from PROVIDES (it's now provided in PROVIDES_devel) Move cmd:ecl-config and it's manpage to the devel package also (there is no pkgconfig file, so the fixPkgconfig can be removed also)
> 

i have moved to the devel package cmd:ecl-config to the devel package although i am not sure how to move the manpage there.

> Is the cmd:ecl actually used or is the library the main use? If so maybe move the cmd:ecl to a seperate tools packge?

the ecl command is used as far as i know. i checked a packaged ecl on a linux install and confirmed that the ecl command is also packaged there as well.

File: dev-lisp/ecl/ecl-24.5.10.recipe (hunk 2 above)
Comment (Begasus) [reply dev-lisp/ecl/ecl-24.5.10.recipe:84]: To move the config and manpage (still had the changes on the 32bit install):

```
	packageEntries devel \
		$developDir \
		$binDir/ecl-config \
		$manDir/man1/ecl-config.1
```

Comment (hoanga) [reply dev-lisp/ecl/ecl-24.5.10.recipe:86]: updated. thank you!

**Comment by Begasus** [comment]
Thanks!

**Pull request merged**